### ISSUE TITLE
Vim keybinds don’t work with Czech keyboard layout

### ISSUE THREAD

**Issue by Cellane** [zed-industries/community#1981]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

This issue might be similar to #1973, #1913, #1698 and possibly others, but I think perhaps filling separately might be beneficial (apologies if wrong!) because my issue is specific to Zed’s vim mode.

With the latest 0.102.1 update, I noticed in the changelog that `g{$,^}` and others should now be supported and tried testing it. However, these don’t seem to actually work when the Czech keyboard (specifically, Czech-QWERTY) is selected as the active layout.

With the Czech keyboard layout, the dollar sign `$` is not input as <kbd>Shift</kbd>-<kbd>4</kbd> (as it is on the US keyboard layout) but rather <kbd>Option</kbd>-<kbd>4</kbd>. Similarly, the caret sign `^` is entered with <kbd>Option</kbd>-<kbd>6</kbd>.

Neither <kbd>Shift</kbd>-<kbd>4</kbd>, <kbd>Option</kbd>-<kbd>4</kbd> or even <kbd>4</kbd> alone seems to move the cursor to the end of the line with the Czech keyboard layout. With the US keyboard layout, the first mentioned keyboard shortcut works as expected.

I think Zed should be reacting on characters actually entered as adjusted by the active keyboard layout, as other editors (incl. vim itself) do.

### Environment

Zed: v0.102.1 (stable)
OS: macOS 13.3.1
Memory: 16 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by ConradIrwin** [comment]
@Cellane Thanks for bringing this up! I would like to work on a fix for everyone, but it's deep inside the keyboard handling code so won't be immediate.

The cause of the problem is that zed treats option in the same way it treats ctrl, so when you type `option-4` it sees `option-4` and not `$`.

Just to confirm: I assume the existing `^` and `$` shortcuts also don't work for you?

Although this is not a great fix, you should be able to work around this by creating custom mappings in your keyboard file, e.g.
```
[
  {
    "context": "Editor && VimControl && !VimWaiting && !menu",
    "bindings": {
      "option-4": "vim::EndOfLine",
      "g option-4": [
        "vim::EndOfLine",
        {
          "displayLines": true
        }
      ],
// etc. etc.
  }
]
```

**Comment by ConradIrwin** [comment]
Sorry for the alt/option confusion, that was my mistake! Glad to hear it is working for you now, and as I said I'll try and work on a fix for everyone.

**Comment by ConradIrwin** [comment]
@Cellane The latest Zed preview (v0.104 https://zed.dev/releases/preview) should fix this so that you can remove your new bindings.

I am not used to working on a Czech keyboard though, so if you are open to testing it, I would love to know if this works as expected or if it breaks something unexpected.

**Comment by ConradIrwin** [comment]
Closing this as fixed now that v0.104 is out. Please let me know if you run into other problems.
